### DISCUSSION TITLE
(Maybe very stupid) Startup Question

### DISCUSSION THREAD

**Discussion by kellertuer** [doconce/doconce#244] [Q&A] [answered]
I have a maybe very stupid startup question, that I can not find much on in the Readme or the docs.

I have installed DocOnce as described in the Readme.
I also have a set of notes from a colleague with a (shell) script that should render that.
But that script calls `doconce`, to be precise the script has the file as its first argument

```
doconce format pdflatex $1 CODE="no" "--latex_code_style=default:lst[style=graycolor]" --latex_preamble=latex_definitions "--latex_admon_color=blue!5"
```

but then I get `command DocOnce`not found.
So. If I followed the installation on the Reamde. How do I get to call doconce and how can I maybe even get a `doconce` command on  shell?

If that is relevant, I am on a Mac.

**Comment by kellertuer** [comment]
To maybe be a little more precise, when I do 

```
❯ pip3 install DocOnce
Requirement already satisfied: DocOnce in ~.local/lib/python3.8/site-packages (1.5.13)
```
(and all following once as well). So I seem to have DocOnce installed. But I can not find how to then call it? And after all reading, I already feel a little stupid, actually.

**Comment by lexnederbragt** [comment]
Hi @kellertuer. Thanks for your interest in using DocOnce!

No worries, installing and using software can be a bit difficult at times. Let me try to help you troubleshooting with some follow-up questions:

Are you using a MAC, Windows PC, Linux machine,...? 

What program do you use to run commands in (usually calls a terminal)?

If you haven't tried it yet, could you run this command in your terminal?

```
doconce --version
```

The fact that the error message is `command DocOnce not found` may help. Is the script called `DocOnce` (with a capital D and O)? If not, what is its name? 

Could you include a screenshot of the terminal window? (it is possible to paste an image in a comment and GitHub will upload it and render it in your comment).

Let's hope the answers to these questions will be enough to solve your problems!

**Reply by kellertuer** [reply]
HI, thanks for the follow up. Its Mac OS (Ventura 13.0.1) and actually both yield the same result

```

~
❯ DocOnce --version
zsh: command not found: DocOnce

~
❯ doconce --version
zsh: command not found: doconce

```
What would a screenshot of this copied text help?

The script is actually called `run_all_notes.sh` but I can read shell code and the first command is along the lines of `doconce myfile.do.txt` (which is an existing file, and yes the command is not capitalised in the script).

**Comment by lexnederbragt** [comment]
OK, that helps. No need for a screenshot, no. What does this do:

```
echo $PATH
```

**Reply by kellertuer** [reply]
Oh, that is a little chaotic already, but I did not see any place where I should add a doconce folder then? Here's my current one

```
❯ echo $PATH
:/opt/homebrew/opt/gnused/libexec/gnubin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/MacGPG2/bin:/Library/Apple/usr/bin:~/bin:/opt/homebrew/opt/fzf/bin
```

**Comment by lexnederbragt** [comment, marked as answer]
OK. I did a little googling and that may help you. What does this give:

```
ls ~/.local/bin
```

If this folder has the doconce command, you could try

```
export PATH="$HOME/.local/bin:$PATH"
```

Then you can try `doconce --version` again.

If this works, I can tell you why it works and how to make it the default setup.

**Reply by kellertuer** [reply]
Oh, I know why that worked

```
~
❯ ls ~/.local/bin
doconce		mako-render	preprocess	publish

~
❯ export PATH="$HOME/.local/bin:$PATH"

~
❯ doconce --version
DocOnce version 1.5.13 (from /Users/ronnber/.local/lib/python3.8/site-packages/doconce)
```

but I would never have found that myself. Nor would I have known where to look for that. I will add that folder to my path in the `.zshenv` and that now also works.

Could that maybe be added somewhere in the install section of the `Readme`?

**Reply by kellertuer** [reply]
And I am aware that this is probably Mac specific (but problely a path is there as well for Linux to set), just that this is not mentioned in the setup (nor could I find it in the manual).

**Comment by lexnederbragt** [comment]
Great that we got it solved! I'll think about how to handle this in the installation instructions (it is not a DocOnce specific thing and I'd rather point people to a general resource on troubleshooting installations - but I have to find that resource first). Follow up in #245.

**Reply by kellertuer** [reply]
Sure, it's a more generic problem, that I was not aware where (its maybe home-brew in this case or pip?) Installs binaries. Still, the installation instruction just stopped at the point where nothing worked for me yet and I was lost (already two weeks ago, but now I actually need doc once).

**Comment by lexnederbragt** [comment]
By the way, would you be willing to describe your usage of DocOnce as a comment to https://github.com/doconce/doconce/discussions/59? That would be appreciated.

**Reply by kellertuer** [reply]
Hi, thanks I will think about that, but it might not be that useful. Mainly: We have notes written in doconce the recent years, that I want to keep updated / corrected if necessary. So indeed, It is not that active usage, more like maintenance.
